orttraining/orttraining/test/python/orttraining_test_ort_apis_onnxblock.py

```diff
@@ -1099,3 +1103,85 @@ def test_custom_optimizer_block():
                 for attr in node.attribute:
                     if attr.name == "weight_decay":
                         assert attr.f == weight_decay
+
+
+def test_generate_artifacts_path():
+
+    with tempfile.TemporaryDirectory() as temp_dir:
+        _, simple_net = _get_models("cpu", 32, 28, 10, 10)
+
+        requires_grad_params = ["fc1.weight", "fc1.bias", "fc2.weight", "fc2.bias"]
+
+        onnx.save_model(
+            simple_net,
+            os.path.join(temp_dir, "simple_net.onnx"),
+        )
+
+        artifacts.generate_artifacts(
+            os.path.join(temp_dir, "simple_net.onnx"),
+            requires_grad=requires_grad_params,
+            loss=artifacts.LossType.CrossEntropyLoss,
+            optimizer=artifacts.OptimType.AdamW,
+            artifact_directory=temp_dir,
+        )
+
+        assert os.path.exists(os.path.join(temp_dir, "training_model.onnx"))
+        assert os.path.exists(os.path.join(temp_dir, "eval_model.onnx"))
+        assert os.path.exists(os.path.join(temp_dir, "optimizer_model.onnx"))
+        assert os.path.exists(os.path.join(temp_dir, "checkpoint"))
```

Comment:
Does generate_artifacts not return an error or raise an exception if all these are not created?

Reply:
Hmm, it usually emits an error if it's not successful, but I thought that I'd add the check just in case -- I can remove it if it's redundant though

Reply:
If generate_artifacts raises an exception as expected, please add a comment to that effect before the asserts. 

```suggestion
        // generate_artifacts should have thrown if it didn't complete successfully. sanity check to validate all expected outputs were created.
        assert os.path.exists(os.path.join(temp_dir, "training_model.onnx"))
        assert os.path.exists(os.path.join(temp_dir, "eval_model.onnx"))
        assert os.path.exists(os.path.join(temp_dir, "optimizer_model.onnx"))
        assert os.path.exists(os.path.join(temp_dir, "checkpoint"))
```

Without that comment, code like this can be misleading. If I read this test without the comment I would assume I needed to add similar checks in my usage, spreading the misconception that generate_artifacts may leave things in an invalid state without warning.